### PR TITLE
Dums 999 Server changes for the Chat

### PR DESCRIPTION
**Pull request by MarcoGoC**
### Description
List of proposed changes:
- Updated server to pass pubsub as part of the context
- Message dataset along with schema, and resolvers 

### What to test for/How to test
- download and run. 
- open apollo studio 
- execute the subscription
- add a message  

### Additional Notes
- Edited and Delete Messages still under development

### CODE REVIEW

File: api/message/subscriptions.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+//
+import { messages } from "../../datasets/messages.js"
+
+export default {
+  messages: {
+    subscribe: (parent, args, { pubsub }) => {
```
Comment (DaceyTom2) [inline api/message/subscriptions.js:6]: A filter can be added to the subscribe so messages are only shared in one room

File: datasets/messages.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  return messages;
+}
+
+function getMessageByID(id) {
+  return messages.find((message) => message.id == id);
+}
+
+//Adds a message
+function addMessage(roomId, name, content) {
+  let id = new Date().getUTCMilliseconds() + Math.random().toString(10).slice(-5);
```
Comment (DaceyTom2) [inline datasets/messages.js:30]: cool idea to use date as an id

**Review by DaceyTom2 — DISMISSED** [review]
Looks really good!

File: api/message/subscriptions.js (hunk 1 above)
Comment (MarcoGoC) [reply api/message/subscriptions.js:6]: I tried and so far i could not make it work. it requires a rethought of everything. 
withFilter requires the single data item, and the second parameter is a Boolean that specifies whether the data should be posted or not. 
the issue here is the messages is an array that contains messages for potentially different rooms, as mention the whole  thing needs to be changed to use withFilter

Comment (DaceyTom2) [reply api/message/subscriptions.js:6]: Sounds good!

**Review by DaceyTom2 — APPROVED** [review]

**Pull request merged**